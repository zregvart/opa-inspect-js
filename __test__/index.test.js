const fs = require('fs');
const path = require('path');
const opa = require('../index');

beforeAll(() => {
    require('events').EventEmitter.defaultMaxListeners = 10;
})

afterAll(() => {
    expect(process.listeners('exit')).toHaveLength(0);
})

test('inspects rego files', async () => {
    const rego = fs.readFileSync(path.join(__dirname, 'example.rego')).toString();
    const json = await opa.inspect('example.rego', rego);
    expect(json).toMatchSnapshot();
});

test('inspects rego files second time', async () => {
    const rego = fs.readFileSync(path.join(__dirname, 'example.rego')).toString();
    const json = await opa.inspect('example.rego', rego);
    expect(json).toMatchSnapshot();
});

test('reports error when parsing', async () => {
    await expect(opa.inspect('example.rego', "bogus")).rejects.toEqual("1 error occurred: example.rego:1: rego_parse_error: package expected");
});

test('inspects rego files read from the filesystem', async () => {
    const json = await opa.inspect(path.join(__dirname, 'example.rego'));
    expect(json).toHaveLength(1);
    expect(json[0]).toMatchSnapshot({
        "location": {
            "file": expect.stringMatching(/.*__test__\/example\.rego/),
        }
    });
});

test('inspects multiple rego files', async () => {
    const json = await opa.inspect([path.join(__dirname, 'example.rego'), path.join(__dirname, 'example2.rego')]);
    expect(json).toHaveLength(2);
    expect(json[0]).toMatchSnapshot({
        "location": {
            "file": expect.stringMatching(/.*__test__\/example\.rego/),
        }
    });
    expect(json[1]).toMatchSnapshot({
        "location": {
            "file": expect.stringMatching(/.*__test__\/example2\.rego/),
        }
    });
});

test('inspects vinyl streams', async () => {
    const vfs = require('vinyl-fs');

    const json = await opa.inspect(vfs.src(path.join(__dirname, '*.rego')));

    expect(json).toHaveLength(2);
    expect(json[0]).toMatchSnapshot({
        "location": {
            "file": expect.stringMatching(/.*__test__\/example\.rego/),
        }
    });
    expect(json[1]).toMatchSnapshot({
        "location": {
            "file": expect.stringMatching(/.*__test__\/example2\.rego/),
        }
    });
});

test.concurrent.each([...Array(25).keys()])('functions concurrently (%#)', async () => {
    const rego = fs.readFileSync(path.join(__dirname, 'example.rego')).toString();
    await expect(opa.inspect('example.rego', rego)).resolves.toHaveProperty('[0].annotations');
});
