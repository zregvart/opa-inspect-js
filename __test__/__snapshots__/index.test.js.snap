// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inspects multiple rego files 1`] = `
[
  {
    "annotations": {
      "custom": {
        "failure_msg": "Task '%s' does not contain a bundle reference",
        "short_name": "disallowed_task_reference",
      },
      "description": "Check for existence of a task bundle. Enforcing this rule will
fail the contract if the task is not called from a bundle.",
      "scope": "rule",
      "title": "Task bundle was not used or is not defined",
    },
    "location": {
      "col": 1,
      "file": StringMatching /\\.\\*__test__\\\\/example\\\\\\.rego/,
      "row": 12,
    },
    "path": [
      {
        "type": "var",
        "value": "data",
      },
      {
        "type": "string",
        "value": "example",
      },
      {
        "type": "string",
        "value": "deny",
      },
    ],
  },
  {
    "annotations": {
      "custom": {
        "short_name": "example2",
      },
      "description": "Second example",
      "scope": "rule",
      "title": "Example 2",
    },
    "location": {
      "col": 1,
      "file": StringMatching /\\.\\*__test__\\\\/example2\\\\\\.rego/,
      "row": 8,
    },
    "path": [
      {
        "type": "var",
        "value": "data",
      },
      {
        "type": "string",
        "value": "example2",
      },
      {
        "type": "string",
        "value": "deny",
      },
    ],
  },
]
`;

exports[`inspects rego files 1`] = `
[
  {
    "annotations": {
      "custom": {
        "failure_msg": "Task '%s' does not contain a bundle reference",
        "short_name": "disallowed_task_reference",
      },
      "description": "Check for existence of a task bundle. Enforcing this rule will
fail the contract if the task is not called from a bundle.",
      "scope": "rule",
      "title": "Task bundle was not used or is not defined",
    },
    "location": {
      "col": 1,
      "file": "example.rego",
      "row": 12,
    },
    "path": [
      {
        "type": "var",
        "value": "data",
      },
      {
        "type": "string",
        "value": "example",
      },
      {
        "type": "string",
        "value": "deny",
      },
    ],
  },
]
`;

exports[`inspects rego files read from the filesystem 1`] = `
[
  {
    "annotations": {
      "custom": {
        "failure_msg": "Task '%s' does not contain a bundle reference",
        "short_name": "disallowed_task_reference",
      },
      "description": "Check for existence of a task bundle. Enforcing this rule will
fail the contract if the task is not called from a bundle.",
      "scope": "rule",
      "title": "Task bundle was not used or is not defined",
    },
    "location": {
      "col": 1,
      "file": StringMatching /\\.\\*__test__\\\\/example\\\\\\.rego/,
      "row": 12,
    },
    "path": [
      {
        "type": "var",
        "value": "data",
      },
      {
        "type": "string",
        "value": "example",
      },
      {
        "type": "string",
        "value": "deny",
      },
    ],
  },
]
`;

exports[`inspects rego files second time 1`] = `
[
  {
    "annotations": {
      "custom": {
        "failure_msg": "Task '%s' does not contain a bundle reference",
        "short_name": "disallowed_task_reference",
      },
      "description": "Check for existence of a task bundle. Enforcing this rule will
fail the contract if the task is not called from a bundle.",
      "scope": "rule",
      "title": "Task bundle was not used or is not defined",
    },
    "location": {
      "col": 1,
      "file": "example.rego",
      "row": 12,
    },
    "path": [
      {
        "type": "var",
        "value": "data",
      },
      {
        "type": "string",
        "value": "example",
      },
      {
        "type": "string",
        "value": "deny",
      },
    ],
  },
]
`;
